src/pathfinder/water_regions.cpp

```diff
@@ -267,8 +269,10 @@ WaterRegionPatchDesc GetWaterRegionPatchInfo(TileIndex tile)
 void InvalidateWaterRegion(TileIndex tile)
 {
 	const int index = GetWaterRegionIndex(tile);
-	if (index > static_cast<int>(_water_regions.size())) return;
+	if (index >= static_cast<int>(_water_regions.size())) return;
```

Comment:
When does this condition occur? If never and it's a safety check then this should probably be an assert (or asserting that tile is valid)?

Reply:
I think this is a remnant of the time where the initialization code was poorly timed. I remember that abandoning a game with a map larger than the menu game issues, but they no longer seem to occur. I will remove this check, bounds checking is already compiled into the [] operator in debug.